s4e-web/src/app/views/settings/state/institution/institution.utils.ts

```diff
@@ -0,0 +1,58 @@
+import { InstitutionStore } from './institution.store';
+import { Institution } from './institution.model';
+import { AkitaGuidService } from 'src/app/views/map-view/state/search-results/guid.service';
+import { InstitutionQuery } from './institution.query';
+import { ActivatedRoute } from '@angular/router';
+import { filter, map } from 'rxjs/operators';
+
+export function saveIn(store: InstitutionStore, institutions: Institution[], generator: AkitaGuidService) {
+  const institutionsWithIds = institutions
+    .map(institution => ({...institution, id: generator.guid()}));
+  store.set(institutionsWithIds);
+}
+
+export function getInstitutionsFrom$(query: InstitutionQuery) {
+  return query.selectAll()
+    .pipe(
+      filter(institutions => {
+        if (institutions.length === 0) {
+          this.store.setError('no_institution');
+        }
+
+        return institutions.length > 0;
+      })
+    );
+}
+
+export function getInstitutionSlugFrom$(route: ActivatedRoute) {
+  return route.queryParamMap
+    .pipe(map(params => params.get('institution')));
+}
+
+export function initiateActiveSlugOnEmpty(route: ActivatedRoute, params) {
+  const [institutions, activeInstitutionSlug] = params;
+  const institutionSlug = !!activeInstitutionSlug
+    ? activeInstitutionSlug
+    : institutions.shift().slug;
+  if (!activeInstitutionSlug) {
+    this.setInstitution(route, institutionSlug);
+  }
+
+  const slugIndex = 1;
+  params[slugIndex] = institutionSlug;
+  return params;
+}
+
+export function setActiveInstitutionSlug(store: InstitutionStore, params) {
+  const [institutions, activeInstitutionSlug] = params;
+  const institutionExists = institutions
+    .some(institution => institution.slug.indexOf(activeInstitutionSlug) > -1);
+  if (!institutionExists) {
+    store.setError('no_institution');
+    return null;
+  }
+
+  store.setActive(activeInstitutionSlug);
+  store.setError(null);
+  return activeInstitutionSlug;
+}
```

Comment:
I'm strongly against moving this out of the service class. I can understand will to make classes less bloated, but this does not solve any problem and only moves code around and makes it harder to find.

Also - are those functions working correctly? I see calls to `this.store` and other `this.*`